internal/resources/providers/gcplib/inventory/provider.go

```diff
@@ -113,36 +113,37 @@ func newAssetsInventoryWrapper(ctx context.Context, log *clog.Logger, gcpConfig
 }
 
 func (p *ProviderInitializer) Init(ctx context.Context, log *clog.Logger, gcpConfig auth.GcpFactoryConfig, cfg config.GcpConfig) (ServiceAPI, error) {
-	assetsInventory, err := newAssetsInventoryWrapper(ctx, log, gcpConfig, cfg)
+	inv, err := newAssetsInventoryWrapper(ctx, log, gcpConfig, cfg)
 	if err != nil {
 		return nil, err
 	}
 
-	cloudResourceManager, err := NewResourceManagerWrapper(ctx, log, gcpConfig)
+	crm, err := NewResourceManagerWrapper(ctx, log, gcpConfig)
 	if err != nil {
 		return nil, err
 	}
 
 	return &Provider{
 		config:    gcpConfig,
 		log:       log,
-		inventory: assetsInventory,
-		crm:       cloudResourceManager,
+		inventory: inv,
+		crm:       crm,
 	}, nil
 }
 
 func (p *Provider) ListAssetTypes(ctx context.Context, assetTypes []string, out chan<- *ExtendedGcpAsset) {
 	defer close(out)
 
-	resourceAssetsCh := p.fetchAssets(ctx, assetpb.ContentType_RESOURCE, assetTypes)
-	policiesAssetsCh := p.fetchAssets(ctx, assetpb.ContentType_IAM_POLICY, assetTypes)
-	assetsCh := p.mergeAssets(ctx, resourceAssetsCh, policiesAssetsCh)
-
-	for asset := range assetsCh {
-		select {
-		case <-ctx.Done():
+	for _, t := range assetTypes {
+		if ctx.Err() != nil {
 			return
-		case out <- asset:
+		}
+
+		resCh := p.getAssets(ctx, p.config.Parent, []string{t}, assetpb.ContentType_RESOURCE)
```

Comment:
nit: It makes more sense to send only the type and append it to a list inside `getAssets` logic.
```suggestion
		resCh := p.getAssets(ctx, p.config.Parent, t, assetpb.ContentType_RESOURCE)
```

Reply:
it would, but `getAssets` is also used elsewhere that requires passing an array.

Reply:
Didn't find a place that sends more than one type, did I miss it?